src/txn/txn.c

```diff
@@ -1213,10 +1214,14 @@ __wt_txn_rollback(WT_SESSION_IMPL *session, const char *cfg[])
 			 * Need to resolve indirect references of transaction
 			 * operation, in case of prepared transaction.
 			 */
-			if (F_ISSET(txn, WT_TXN_PREPARE))
-				WT_RET(__wt_txn_resolve_prepared_op(
-				    session, op, false));
-			else {
+			if (F_ISSET(txn, WT_TXN_PREPARE)) {
+				if (!F_ISSET(op, WT_TXN_OP_KEY_REPEATED))
+					WT_RET(__wt_txn_resolve_prepared_op(
```

Comment:
we are skipping over the txn op, if we haven't found a prepared update, which should not happen.